client-mr/src/main/java/org/apache/hadoop/mapreduce/task/reduce/RssEventFetcher.java

```diff
@@ -73,18 +73,21 @@ public Roaring64NavigableMap fetchAllRssTaskIds() {
     String errMsg = "TaskAttemptIDs are inconsistent with map tasks";
     for (TaskAttemptID taskAttemptID: successMaps) {
       if (!obsoleteMaps.contains(taskAttemptID)) {
-        long rssTaskId = RssMRUtils.convertTaskAttemptIdToLong(taskAttemptID);
-        taskIdBitmap.addLong(rssTaskId);
         int mapIndex = taskAttemptID.getTaskID().getId();
-        if (mapIndex < totalMapsCount) {
-          mapIndexBitmap.addLong(mapIndex);
-        } else {
-          throw new IllegalStateException(errMsg);
+        // we only need to just one attempt of each map task.
```

Comment:
one successful attempt?

Reply:
This logic is to skip redundancy successful attempt.

Otherwise   
if (taskIdBitmap.getLongCardinality() + tipFailedCount != totalMapsCount) {
      throw new IllegalStateException(errMsg);
}

Reply:
I mean you should modify the comment.

Reply:
Updated